### PR TITLE
experimental: add async HydSim methods

### PR DESCRIPTION
**Pull request by erny-powel**
Includes hydro simulation and inflow calculation. PQ curve generation will be added separately when server support is available. This API is very much subject to change for the near term future.

See https://github.com/Volue/energy-sim/issues/742.

Closes https://github.com/Volue-Public/energy-mesh-python/issues/395.

### CODE REVIEW

File: src/volue/mesh/aio/_connection.py [hunk 1]
```diff
@@ -379,6 +381,64 @@ async def update_rating_curve_versions(
...
+
+            simulation = f"Model/{model}/{case_group}.has_OptimisationCases/{case_name}.has_OptimisationParameters/Optimal.has_HydroSimulation/HydroSimulation"
+
+            request = core_pb2.SimulationRequest(
+                session_id=_to_proto_guid(self.session_id),
+                simulation=_to_proto_object_mesh_id(simulation),
+                interval=_to_proto_utcinterval(start_time, end_time),
+                scenario=scenario,
+                return_datasets=return_datasets,
+            )
```
Comment (tnoczyns-volue) [inline src/volue/mesh/aio/_connection.py:407]: You can move the code for checking timestamps, splitting case group and name and request preparation to some `_prepare_simulation_request`. We are doing something like this for some other RPCs. I'm all for not repeating the code if possible, even if it means to go one level deeper to find out the logic. The same for inflow calculation. Not a blocker though.

File: src/volue/mesh/examples/run_inflow_calculation_async.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import asyncio
+from datetime import datetime
+
+from volue import mesh
+import volue.mesh.aio
+
+from volue.mesh.examples import _get_connection_info
+
+
+async def main(address, port, root_pem_certificate):
```
Comment (tnoczyns-volue) [inline src/volue/mesh/examples/run_inflow_calculation_async.py:10]: Just wanted to mention we are not too consistent in the sync vs async examples. Some times we create separate files and some time swe do it within one file, like in `authorization.py` or `get_version.py`. Personally I like it to be in one file. (possibly more re-use of parameters, and we have the functionality example in one place). Not a blocker, just mentioning.

**Review by tnoczyns-volue — APPROVED** [review]

File: src/volue/mesh/aio/_connection.py (hunk 1 above)
Comment (erny-powel) [reply src/volue/mesh/aio/_connection.py:407]: Will resolve in a follow-up PR, I'm planning to do the `_base_session.py` thing as well.

File: src/volue/mesh/examples/run_inflow_calculation_async.py (hunk 2 above)
Comment (erny-powel) [reply src/volue/mesh/examples/run_inflow_calculation_async.py:10]: Ah, nice! I just looked at the files. I'll do the one file thing then, makes things a bit easier.

**Comment by erny-powel** [comment]
Thanks for the review!

**Pull request merged**